addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -49,12 +50,23 @@ odoo.define('website.s_website_form', function (require) {
             this._recaptcha = new ReCaptcha();
             this.__started = new Promise(resolve => this.__startResolve = resolve);
         },
-        willStart: function () {
+        willStart: async function () {
             const res = this._super(...arguments);
             if (!this.$target[0].classList.contains('s_website_form_no_recaptcha')) {
                 this._recaptchaLoaded = true;
                 this._recaptcha.loadLibs();
             }
+            // fetch user data (required by pre-fill option)
+            this.userData = {};
+            this.canBePreFilled = session.user_id && this.$target[0].dataset.preFill === 'true';
+            if (this.canBePreFilled) {
+                this.userData = (await this._rpc({
+                    model: 'res.users',
+                    method: 'read',
+                    args: [session.user_id, ['name', 'phone', 'email', 'commercial_company_name']],
```

Comment:
Shouldn't this list of field be inside a function?
Imagine we want to add a website_sale related field (not existing in website) inside the form later (or the community), this module would need to be able to overide that list no?
That would be enough for it to work, right? Add field + data-fill-with in FormRegistry then txtend this list to read that field?